cloudflare/resource_cloudflare_worker_secret_test.go

```diff
@@ -0,0 +1,165 @@
+	t.Parallel()
+
+	name := generateRandomResourceName()
+	secretText := generateRandomResourceName()
+	workerSecretTestScriptName = generateRandomResourceName()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheckAccount(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareWorkerSecretDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareWorkerSecretWithWorkerScript(workerSecretTestScriptName, name, secretText),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareWorkerSecretExists(workerSecretTestScriptName, name, secretText),
+				),
+			},
+		},
+	})
+
+	deleteWorker()
+}
+
+func testAccCheckCloudflareWorkerSecretDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*cloudflare.API)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "cloudflare_worker_secret`" {
+			continue
+		}
+
+		params := getRequestParamsFromResource(rs)
+		secretResponse, err := client.ListWorkersSecrets(context.Background(), params.ScriptName)
+
+		if err != nil {
+			return err
+		}
+
+		if len(secretResponse.Result) > 0 {
+			return fmt.Errorf("Worker secret with name %s still exists against Work Script %s", secretResponse.Result[0].Name, params.ScriptName)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckCloudflareWorkerSecretWithWorkerScript(scriptName string, name string, secretText string) string {
+	err := createWorker()
+
+	if err != nil {
+		panic(err)
+	}
+
+	return fmt.Sprintf(`
+	resource "cloudflare_worker_secret" "%[2]s" {
+		script_name = "%[1]s"
+		name 		= "%[2]s"
+		secret_text	= "%[3]s"
+	}`, scriptName, name, secretText)
+}
+
+func testAccCheckCloudflareWorkerSecretExists(scriptName string, name string, secretText string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := testAccProvider.Meta().(*cloudflare.API)
+		secretResponse, err := client.ListWorkersSecrets(context.Background(), scriptName)
+		if err != nil {
+			return err
+		}
+
+		for _, secret := range secretResponse.Result {
+			if secret.Name == name {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("Worker secret with name %s not found against Worker Script %s", name, scriptName)
+	}
+}
+
+func createWorker() error {
+	client, err := createCloudflareClient()
+
+	if err != nil {
+		panic(err)
+	}
+
+	scriptRequestParams := cloudflare.WorkerRequestParams{
+		ScriptName: workerSecretTestScriptName,
+	}
+
+	_, err = client.UploadWorker(&scriptRequestParams, scriptContentForSecret)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func deleteWorker() error {
+	client, err := createCloudflareClient()
+
+	if err != nil {
+		return err
+	}
+
+	scriptRequestParams := cloudflare.WorkerRequestParams{
+		ScriptName: workerSecretTestScriptName,
+	}
+
+	_, err = client.DeleteWorker(&scriptRequestParams)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createCloudflareClient() (*cloudflare.API, error) {
```

Comment:
we don't need to re-create an API client, the provider test harness does this out of the box - you will need to provide it with the correct credentials via environment variables though.

Reply:
Gotcha, that to me is an indication that we're holding something wrong as the provider credential configuration should be the entry point for all test cases. Let's continue this in the comment above as I think that will unlock some more ideas.